app/src/main/java/com/example/moviedb/data/service/api/RetrofitInstance.kt

```diff
@@ -0,0 +1,22 @@
+package com.example.moviedb.data.service.api
+
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+
+class RetrofitInstance {
```

Comment:
sử dụng object 
từ khóa object trong Kotlin = class no param constructor + singleton 
tức là, nếu ta khai báo 
```suggestion
object A {
    fun execute()     
}
```
và gọi `A.execute()`
thì nó sẽ tương ứng trong java 
`A.getInstance().execute() `
```suggestion
object ApiFactory {

    fun <T> buildRestApi(
        baseUrl: String,
        restApi: Class<T>,
        converterFactory: GsonConverterFactory,
        callAdapterFactory: CoroutineCallAdapterFactory
    ): T =
        Retrofit.Builder().baseUrl(baseUrl)
            .addConverterFactory(converterFactory)
            .addCallAdapterFactory(callAdapterFactory)
            .build()
            .create(restApi)
}
```